diff --git a/node_modules/@react-native-community/art/ReactNativeART.podspec b/node_modules/@react-native-community/art/ReactNativeART.podspec
index 9ab4b45..a0fb152 100644
--- a/node_modules/@react-native-community/art/ReactNativeART.podspec
+++ b/node_modules/@react-native-community/art/ReactNativeART.podspec
@@ -20,5 +20,5 @@ Pod::Spec.new do |s|
   s.source                 = { :git => "https://github.com/react-native-community/art.git", :tag => "v#{s.version}" }
   s.source_files           = "ios/**/*.{h,m}"
 
-  s.dependency "React"
+  s.dependency 'React-Core'
 end
