diff --git a/node_modules/multiserver/plugins/net.js b/node_modules/multiserver/plugins/net.js
index ff1b663..dec9e02 100644
--- a/node_modules/multiserver/plugins/net.js
+++ b/node_modules/multiserver/plugins/net.js
@@ -16,7 +16,7 @@ module.exports = function (opts) {
   return {
     name: 'net',
     server: function (onConnection) {
-      var server = net.createServer(opts, function (stream) {
+      var server = net.createServer(function (stream) {
         onConnection(toDuplex(stream))
       }).listen(opts.port)
       return function () {
