patch-package
--- a/node_modules/nodejs-mobile-react-native/android/src/main/java/com/janeasystems/rn_nodejs_mobile/RNNodeJsMobileModule.java
+++ b/node_modules/nodejs-mobile-react-native/android/src/main/java/com/janeasystems/rn_nodejs_mobile/RNNodeJsMobileModule.java
@@ -45,6 +45,11 @@ public class RNNodeJsMobileModule extends ReactContextBaseJavaModule implements
   private static String builtinModulesPath;
   private static String nativeAssetsPath;

+  public interface Invokable<T> {
+    public void invoke(T param);
+  }
+
+  private static Invokable<String> jsExceptionInvokable;
   private static long lastUpdateTime = 1;
   private static long previousLastUpdateTime = 0;
   private static Semaphore initSemaphore = new Semaphore(1);
@@ -91,6 +96,10 @@ public class RNNodeJsMobileModule extends ReactContextBaseJavaModule implements
     asyncInit();
   }

+  public void setJsExceptionInvokable(Invokable<String> i) {
+    jsExceptionInvokable = i;
+  }
+
   private void asyncInit() {
     if (wasAPKUpdated()) {
       try {
@@ -243,6 +236,12 @@ public class RNNodeJsMobileModule extends ReactContextBaseJavaModule implements
       final RNNodeJsMobileModule _moduleInstance = _instance;
       final String _channelNameToPass = new String(channelName);
       final String _msgToPass = new String(msg);
+      if (jsExceptionInvokable != null) {
+        if (_msgToPass.matches("^\\{\"event\":\"exception\",\"payload\":\".*")) {
+          final String _exception = _msgToPass.substring(35, _msgToPass.length() - 5);
+          jsExceptionInvokable.invoke(_exception);
+        }
+      }
       new Thread(new Runnable() {
         @Override
         public void run() {
