diff --git a/node_modules/react-native-bluetooth-socket-bridge/android/src/main/AndroidManifest.xml b/node_modules/react-native-bluetooth-socket-bridge/android/src/main/AndroidManifest.xml
index d7696dd..1fbde2f 100644
--- a/node_modules/react-native-bluetooth-socket-bridge/android/src/main/AndroidManifest.xml
+++ b/node_modules/react-native-bluetooth-socket-bridge/android/src/main/AndroidManifest.xml
@@ -6,9 +6,4 @@
     <uses-permission android:name="android.permission.BLUETOOTH_ADMIN" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
 
-    <uses-sdk
-        android:minSdkVersion="19"
-        android:targetSdkVersion="22" />
-
 </manifest>
-  
\ No newline at end of file
diff --git a/node_modules/react-native-bluetooth-socket-bridge/android/src/main/java/com/scuttlebutt/bluetoothbridge/control/ControlUnixSocket.java b/node_modules/react-native-bluetooth-socket-bridge/android/src/main/java/com/scuttlebutt/bluetoothbridge/control/ControlUnixSocket.java
index b087cfe..dc87ec8 100644
--- a/node_modules/react-native-bluetooth-socket-bridge/android/src/main/java/com/scuttlebutt/bluetoothbridge/control/ControlUnixSocket.java
+++ b/node_modules/react-native-bluetooth-socket-bridge/android/src/main/java/com/scuttlebutt/bluetoothbridge/control/ControlUnixSocket.java
@@ -102,7 +102,9 @@ public class ControlUnixSocket {
         return new Runnable() {
             @Override
             public void run() {
-                LocalSocket localSocket = establishConnection(10);
+                // Run 100000 retries because we dont know when ssb-server
+                // starts, due to user being on welcome screen
+                LocalSocket localSocket = establishConnection(100000);
 
                 Thread responseWriter =  new Thread(responseWriterThread(localSocket));
                 responseWriter.start();
