diff --git a/node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSPackage.java b/node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSPackage.java
index 7cf9575..ceccc74 100644
--- a/node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSPackage.java
+++ b/node_modules/react-native-fs/android/src/main/java/com/rnfs/RNFSPackage.java
@@ -17,11 +17,6 @@ public class RNFSPackage implements ReactPackage {
     return modules;
   }

-  @Override
-  public List<Class<? extends JavaScriptModule>> createJSModules() {
-    return Collections.emptyList();
-  }
-
   @Override
   public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
     return Arrays.<ViewManager>asList();
