diff --git a/node_modules/react-native-system-setting/RCTSystemSetting.podspec b/node_modules/react-native-system-setting/RCTSystemSetting.podspec
index 63c6286..a2c21ed 100644
--- a/node_modules/react-native-system-setting/RCTSystemSetting.podspec
+++ b/node_modules/react-native-system-setting/RCTSystemSetting.podspec
@@ -16,5 +16,5 @@ Pod::Spec.new do |s|
   s.source_files    = 'ios/*.{h,m}'
   s.preserve_paths  = "**/*.js"
 
-  s.dependency 'React'
+  s.dependency 'React-Core'
 end
