diff --git a/node_modules/react-native-gifted-chat/lib/MessageContainer.js b/node_modules/react-native-gifted-chat/lib/MessageContainer.js
index 193772a..6727e92 100644
--- a/node_modules/react-native-gifted-chat/lib/MessageContainer.js
+++ b/node_modules/react-native-gifted-chat/lib/MessageContainer.js
@@ -176,8 +176,7 @@ export default class MessageContainer extends React.PureComponent {
                 distanceFromEnd <= 100 &&
                 loadEarlier &&
                 onLoadEarlier &&
-                !isLoadingEarlier &&
-                Platform.OS !== 'web') {
+                !isLoadingEarlier) {
                 onLoadEarlier();
             }
         };
