diff --git a/node_modules/react-native-fs/RNFS.podspec b/node_modules/react-native-fs/RNFS.podspec
index 74c3727..fac9d65 100644
--- a/node_modules/react-native-fs/RNFS.podspec
+++ b/node_modules/react-native-fs/RNFS.podspec
@@ -18,5 +18,5 @@ Pod::Spec.new do |s|
   s.source_files    = '*.{h,m}'
   s.preserve_paths  = "**/*.js"
 
-  s.dependency 'React'
+  s.dependency 'React-Core'
 end
\ No newline at end of file
