diff --git a/node_modules/react-native-android-statusbar/src/main/java/me/neo/react/StatusBarPackage.java b/node_modules/react-native-android-statusbar/src/main/java/me/neo/react/StatusBarPackage.java
index d148b5c..5ed77b6 100755
--- a/node_modules/react-native-android-statusbar/src/main/java/me/neo/react/StatusBarPackage.java
+++ b/node_modules/react-native-android-statusbar/src/main/java/me/neo/react/StatusBarPackage.java
@@ -26,11 +26,6 @@ public class StatusBarPackage implements ReactPackage {
         return modules;
     }
 
-    @Override
-    public List<Class<? extends JavaScriptModule>> createJSModules() {
-        return Collections.emptyList();
-    }
-
     @Override
     public List<ViewManager> createViewManagers(ReactApplicationContext reactApplicationContext) {
         return Arrays.<ViewManager>asList();
