diff --git a/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js b/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
index 801cc86..e6ed8c5 100644
--- a/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
+++ b/node_modules/react-native-web/dist/vendor/react-native/VirtualizedList/index.js
@@ -671,6 +671,8 @@ class VirtualizedList extends React.PureComponent {
             // we will trust the initialScrollIndex suggestion.
             if (!this.props.initialScrollIndex || this._scrollMetrics.offset) {
               newState = computeWindowedRenderLimits(this.props.data, this.props.getItemCount, maxToRenderPerBatchOrDefault(this.props.maxToRenderPerBatch), windowSizeOrDefault(this.props.windowSize), state, this._getFrameMetricsApprox, this._scrollMetrics);
+              // See https://github.com/necolas/react-native-web/issues/1579
+              newState.first = 0;
             }
           }
         } else {
@@ -882,7 +884,67 @@ class VirtualizedList extends React.PureComponent {
 
   setupWebWheelHandler() {
     if (this._scrollRef && this._scrollRef.getScrollableNode) {
-      this._scrollRef.getScrollableNode().addEventListener('wheel', this.invertedWheelEventHandler);
+      const node = this._scrollRef.getScrollableNode()
+      node.addEventListener('wheel', this.invertedWheelEventHandler);
+      let lastKeyDown = 0;
+      node.addEventListener('keydown', ev => {
+        if (ev.ctrlKey || ev.metaKey) return;
+        const DELTA = 40;
+        const PAGE = node.clientHeight * 0.9;
+        const TOTAL = node.scrollHeight;
+        const behavior = (Date.now() - lastKeyDown) > 60 ? 'smooth' : 'instant';
+        lastKeyDown = Date.now();
+        if (ev.code === 'ArrowDown') {
+          node.scroll({
+            top: node.scrollTop + (this.props.inverted ? -DELTA : +DELTA),
+            left: 0,
+            behavior
+          });
+        } else if (ev.code === 'ArrowUp') {
+          node.scroll({
+            top: node.scrollTop + (this.props.inverted ? +DELTA : -DELTA),
+            left: 0,
+            behavior
+          });
+        } else if (ev.code === 'PageDown') {
+          node.scroll({
+            top: node.scrollTop + (this.props.inverted ? -PAGE : +PAGE),
+            left: 0,
+            behavior
+          });
+        } else if (ev.code === 'PageUp') {
+          node.scroll({
+            top: node.scrollTop + (this.props.inverted ? +PAGE : -PAGE),
+            left: 0,
+            behavior
+          });
+        } else if (ev.code === 'Space' && !ev.shiftKey) {
+          node.scroll({
+            top: node.scrollTop + (this.props.inverted ? -PAGE : +PAGE),
+            left: 0,
+            behavior
+          });
+        } else if (ev.code === 'Space' && ev.shiftKey) {
+          node.scroll({
+            top: node.scrollTop + (this.props.inverted ? +PAGE : -PAGE),
+            left: 0,
+            behavior
+          });
+        } else if (ev.code === 'End') {
+          node.scroll({
+            top: this.props.inverted ? 0 : TOTAL,
+            left: 0,
+            behavior: 'smooth'
+          });
+        } else if (ev.code === 'Home') {
+          node.scroll({
+            top: this.props.inverted ? TOTAL : 0,
+            left: 0,
+            behavior: 'smooth'
+          });
+        }
+        ev.preventDefault();
+      })
     } else {
       setTimeout(() => this.setupWebWheelHandler(), 50);
       return;
