diff --git a/node_modules/rn-viewpager/viewpager/ViewPager.js b/node_modules/rn-viewpager/viewpager/ViewPager.js
index 41f8f1b..0215ffa 100644
--- a/node_modules/rn-viewpager/viewpager/ViewPager.js
+++ b/node_modules/rn-viewpager/viewpager/ViewPager.js
@@ -4,7 +4,8 @@
 
 'use strict'
 
-import { PanResponder, Platform, ScrollView, StyleSheet, View, ViewPagerAndroid } from 'react-native'
+import { PanResponder, Platform, ScrollView, StyleSheet, View } from 'react-native'
+import ViewPagerAndroid from '@react-native-community/viewpager';
 import React, { Component } from 'react'
 
 const SCROLLVIEW_REF = 'scrollView'
