diff --git a/node_modules/jimp/dist/index.js b/node_modules/jimp/dist/index.js
index 1210d81..22a51d3 100644
--- a/node_modules/jimp/dist/index.js
+++ b/node_modules/jimp/dist/index.js
@@ -9,12 +9,10 @@ exports["default"] = void 0;
 
 var _custom = _interopRequireDefault(require("@jimp/custom"));
 
-var _types = _interopRequireDefault(require("@jimp/types"));
-
 var _plugins = _interopRequireDefault(require("@jimp/plugins"));
 
 var _default = (0, _custom["default"])({
-  types: [_types["default"]],
+  types: [],
   plugins: [_plugins["default"]]
 });
 
