diff --git a/node_modules/ssb-db2/core.js b/node_modules/ssb-db2/core.js
index 0950879..87dc101 100644
--- a/node_modules/ssb-db2/core.js
+++ b/node_modules/ssb-db2/core.js
@@ -777,7 +777,9 @@ exports.init = function (sbot, config) {
             batch(1000),
             toPullStream()
           ),
-          pull.asyncMap(log.del),
+          pull.asyncMap((offset, cb) => {
+            setTimeout(() => log.del(offset, cb), 1)
+          }),
           pull.onEnd((err) => {
             // prettier-ignore
             if (err) return cb(new Error('deleteFeed() failed for feed ' + feedId, {cause: err}))
diff --git a/node_modules/ssb-db2/index.js b/node_modules/ssb-db2/index.js
index b98a532..fc277de 100644
--- a/node_modules/ssb-db2/index.js
+++ b/node_modules/ssb-db2/index.js
@@ -6,7 +6,6 @@ module.exports = [
   require('./core'),
   require('ssb-classic'),
   require('ssb-box'),
-  require('ssb-box2'),
   require('./compat/publish'),
   require('./migrate'),
 ]
