diff --git a/node_modules/ssb-keys-neon/native/Cargo.toml b/node_modules/ssb-keys-neon/native/Cargo.toml
index 4e72059..6fc3dbb 100644
--- a/node_modules/ssb-keys-neon/native/Cargo.toml
+++ b/node_modules/ssb-keys-neon/native/Cargo.toml
@@ -4,6 +4,7 @@ version = "8.0.0"
 authors = ["Andre Staltz <andre@staltz.com>"]
 license = "LGPL-3.0"
 build = "build.rs"
+links = "node"
 exclude = ["artifacts.json", "index.node"]
 edition = "2018"
 
