diff --git a/node_modules/ssb-validate2-rsjs/native/Cargo.toml b/node_modules/ssb-validate2-rsjs/native/Cargo.toml
index 38899ed..f1e935e 100644
--- a/node_modules/ssb-validate2-rsjs/native/Cargo.toml
+++ b/node_modules/ssb-validate2-rsjs/native/Cargo.toml
@@ -3,6 +3,7 @@ name = "ssb-validate2-rsjs"
 version = "0.2.0"
 authors = ["Andrew Reid <glyph@mycelial.technology>"]
 edition = "2018"
+links = "node"
 
 [lib]
 crate-type = ["cdylib"]
