diff --git a/node_modules/ssb-conn/lib/conn.js b/node_modules/ssb-conn/lib/conn.js
index 21fb1f8..26aaa7b 100644
--- a/node_modules/ssb-conn/lib/conn.js
+++ b/node_modules/ssb-conn/lib/conn.js
@@ -87,9 +87,9 @@ let CONN = class CONN {
         const that = this;
         this.ssb.close.hook(function (fn, args) {
             that.stopScheduler();
-            that._db.close();
-            that._hub.close();
             that._staging.close();
+            that._hub.close();
+            that._db.close();
             return fn.apply(this, args);
         });
     }
