diff --git a/node_modules/ssb-db2/core.js b/node_modules/ssb-db2/core.js
index 67e1aa5..b487e83 100644
--- a/node_modules/ssb-db2/core.js
+++ b/node_modules/ssb-db2/core.js
@@ -765,7 +765,9 @@ exports.init = function (sbot, config) {
             batch(1000),
             toPullStream()
           ),
-          pull.asyncMap(log.del),
+          pull.asyncMap((offset, cb) => {
+            setTimeout(() => log.del(offset, cb), 1)
+          }),
           pull.onEnd((err) => {
             // prettier-ignore
             if (err) return cb(clarify(err, 'deleteFeed() failed for feed ' + feedId))
diff --git a/node_modules/ssb-db2/index.js b/node_modules/ssb-db2/index.js
index 3205490..f7de23c 100644
--- a/node_modules/ssb-db2/index.js
+++ b/node_modules/ssb-db2/index.js
@@ -6,7 +6,6 @@ module.exports = [
   require('./core'),
   require('ssb-classic'),
   require('ssb-box'),
-  require('ssb-box2'),
   require('./compat/publish'),
   require('./compat/post'),
   require('./migrate'),
