diff --git a/node_modules/ssb-db2/db.js b/node_modules/ssb-db2/db.js
index 1e4220e..df05a63 100644
--- a/node_modules/ssb-db2/db.js
+++ b/node_modules/ssb-db2/db.js
@@ -531,7 +531,9 @@ exports.init = function (sbot, config) {
 
             push(
               push.values(offsets),
-              push.asyncMap(log.del),
+              push.asyncMap((offset, cb) => {
+                setTimeout(() => log.del(offset, cb))
+              }),
               push.collect((err) => {
                 // prettier-ignore
                 if (err) return cb(clarify(err, 'deleteFeed() failed for feed ' + feedId))
